test/e2e/app_secret_configmap_reconcile_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func Test_AppReconcileOccurs_WhenSecretUpdated(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+	sas := ServiceAccounts{env.Namespace}
+
+	name := "configmap-with-secret"
+	// syncPeriod set to 5 minutes so that test
```

Comment:
do you want to crank it up to something high like 60min?

Reply:
Sure, I added.